### DISCUSSION TITLE
Recover custom tags after moving installed folder

### DISCUSSION THREAD

**Discussion by Chubbly** [zanllp/infinite-image-browsing#291] [Q&A] [answered]
I moved my stable-diffusion-webui folder to another location. I didn't realise this would break all the setting for this extension, even which images have been tagged as favourites. I assume the iib.db is storing this information using absolute paths, or something? Is there anyway 

1. Recover the existing "liked" images without moving the folder back to its original location
2. Safely move the installed location so it doesn't lose all tag information?

Any help would be appreciated.

**Comment by zanllp** [comment, marked as answer]
I'm sorry, but currently there is no way to recover the existing "liked" images without moving the folder back to its original location. Different paths will be considered as different files, and the extension will remove any inaccessible records. The reason why absolute paths are used is because the sd_webui path supports them, and I personally prefer to keep all images outside of the sd_webui folder.

If you need to move the installed location in the future, I can write a script to help you modify the records.

**Reply by Chubbly** [reply]
I explored the iib.db file myself as it appears to be a sqllite db. If I wrote some SQL to update the folder and image_path path columns, would that restore my liked images in the new location?

**Reply by zanllp** [reply]
The records in iib.db that were updated to the correct path can indeed be accessed. However, any records that have already been deleted cannot be accessed anymore.

**Reply by Manniala** [reply]
Are there a chance you could provide such a script for me?
I am moving to use "StabilityMatrix" and that one uses another path, so i would lose all my tags.

**Comment by Manniala** [comment]
[Know its old post]
Well for this, there could be implement a hash check, but it should be a optional thing, for instance you run it when you know you have to move a lot of images, it than generates hash for the images, once you are done moving the images, you than run a scan again that finds the matching hash..

But the hash should be in a separate .db file i think, and in global settings you can toggle it on and off, that way it wont have to be loaded when you browse your images (not sure if it would slow down the process).

**Comment by zanllp** [comment]
I have implemented a solution to address this issue, which involves an automated database file backup mechanism and a migration script. For details, please refer to the following link: https://github.com/zanllp/sd-webui-infinite-image-browsing/wiki/Proper-Folder-Migration-to-Avoid-Data-Loss---%E6%AD%A3%E7%A1%AE%E7%9A%84%E8%BF%81%E7%A7%BB%E6%96%87%E4%BB%B6%E5%A4%B9%E9%81%BF%E5%85%8D%E6%95%B0%E6%8D%AE%E4%B8%A2%E5%A4%B1.
